src/Filesystem/File.php

```diff
@@ -368,7 +388,7 @@ public function safe($name = null, $ext = null)
             $ext = $this->ext();
         }
 
-        return preg_replace("/(?:[^\w\.-]+)/", '_', basename($name, $ext));
+        return preg_replace("/(?:[^\w\.-]+)/", '_', static::_basename($name, $ext));
```

Comment:
probably needs a bit test to testing this method 
`public function safe($name = null, $ext = null)`